### PR TITLE
Update mention_channels documentation with constraints

### LINKED ISSUE #792: Mention_channels vec from Message is not returned when there is text in the message.
For some odd reason when trying to get the vec of mention_channels from Message, it will return None if there is other text in the message. If there is no text in the message and only a channel mention, then it will return.

### PR DESCRIPTION
**Pull request by zacck-zz**
## Changes Proposed 

Fixes #792

Add document to detail the features of the mention_channels vec and supply a link to discord with  more information

### CODE REVIEW

File: src/model/channel/message.rs [hunk 1]
```diff
@@ -72,6 +72,12 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    /// Please note:
```
Comment (drklee3) [inline src/model/channel/message.rs]: Minor wording and formatting change to be consistent with other notes in documentation.

```suggestion
    ///
    /// **Note**:
```

File: src/model/channel/message.rs [hunk 2]
```diff
@@ -72,6 +72,12 @@ pub struct Message {
...
+    /// Not all channel mentions in a message will appear in mention_channels.
+    /// Only textual channels that are visible to everyone in a lurkable guild will ever be included.
+    /// Only crossposted messages (via Channel Following) currently include mention_channels at all.
+    /// If no mentions in the message meet these requirements, this field will not be sent.
+    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```
Comment (drklee3) [inline src/model/channel/message.rs]: [Reference-style links](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) need to be on their own line to render properly.

Regular links are like ``[`See more in the discord documentation`](https://discord.com/developers/docs/resources/channel#message-object)`` with both the text and url together, but reference-style links need to have their url defined separate from the text.

```suggestion
    /// [`See more in the discord documentation`].
    ///
    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```

**Review by drklee3 — CHANGES_REQUESTED** [review]
Good clarification 👍  Just some small formatting changes. Also, could you rewrap the text so it is max 80 columns?

**Comment by zacck-zz** [comment]
@drklee3  did some corrections does this look better?

File: src/model/channel/message.rs [hunk 3]
```diff
@@ -72,6 +72,13 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    /// **Note**:
+    /// Not all channel mentions in a message will appear in mention_channels.
+    /// Only textual channels that are visible to everyone in a lurkable guild will ever be included.
+    /// Only crossposted messages (via Channel Following) currently include mention_channels at all.
+    /// If no mentions in the message meet these requirements, this field will not be sent.
+    /// [`See more in the discord documentation`].
+    /// https://discord.com/developers/docs/resources/channel#message-object
```
Comment (drklee3) [inline src/model/channel/message.rs]: You can see examples for reference-style links [here](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links)

```markdown
    /// 👇 This is the link text by itself
    /// [`See more in the discord documentation`].
    ///
    /// 👇 This is where the link is defined which needs a blank line before it.
    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```

```suggestion
    /// [`See more in the discord documentation`].
    ///
    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```

File: src/model/channel/message.rs [hunk 4]
```diff
@@ -72,6 +72,13 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    /// **Note**:
```
Comment (drklee3) [inline src/model/channel/message.rs:76]: Just a small blank line here so the note is on it's own line.  (Markdown needs 2 blank lines to actually make a line break)

```suggestion
    ///
    /// **Note**:
```

**Review by drklee3 — COMMENTED** [review]
Almost there! I would suggest running `cargo doc --no-deps --open` to build the documentation files and then checking the formatting in your browser.

Currently it looks like this:

image

So you can see the link isn't displaying correctly, and the `Note` is all on the same line but we want it on it's own line below the description.

Also a tip: You can use the `Add suggestion to batch` buttons to easily commit all the suggested changes with just a few clicks.

**Comment by zacck-zz** [comment]
@drklee3  Just confirmed, I think this one displays correctly

**Review by drklee3 — APPROVED** [review]
Looks good, thanks!

File: src/model/channel/message.rs [hunk 5]
```diff
@@ -72,6 +72,15 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    ///
+    /// **Note**:
+    /// Not all channel mentions in a message will appear in mention_channels.
```
Comment (Lakelezz) [inline src/model/channel/message.rs]:
```suggestion
    /// Not all channel mentions in a message will appear in `mention_channels`.
```
We refer to fields and other Rust items by `ticks`.

File: src/model/channel/message.rs [hunk 6]
```diff
@@ -72,6 +72,15 @@ pub struct Message {
...
+    /// Only textual channels that are visible to everyone in a lurkable guild will ever be included.
```
Comment (Lakelezz) [inline src/model/channel/message.rs]: I do know this is taken from the official Discord documentation but I personally feel the term *lurkable* is really elusive.
Do you maybe know a term that explains it better? In what way is a guild *lurkable*? I personally do not know what Discord tries to express here.

File: src/model/channel/message.rs [hunk 7]
```diff
@@ -72,6 +72,15 @@ pub struct Message {
...
+    /// Only crossposted messages (via Channel Following) currently include mention_channels at all.
```
Comment (Lakelezz) [inline src/model/channel/message.rs]:
```suggestion
    /// Only crossposted messages (via Channel Following) currently include `mention_channels` at all.
```
As mentioned, Rust items should be `tick'd`.

**Review by Lakelezz — DISMISSED** [review]

File: src/model/channel/message.rs (hunk 6 above)
Comment (zacck-zz) [reply src/model/channel/message.rs]: I actually speaking to them about  some issues will ask

File: src/model/channel/message.rs [hunk 8]
```diff
@@ -72,6 +72,18 @@ pub struct Message {
...
+    /// Not all channel mentions in a message will appear in `mention_channels`.
+    /// Only textual channels that are visible to everyone in a lurkable guild will ever be included.
+    /// A lurkable guild being one that allows users to read public channels in a server without
+    /// actually joining the server. It also allows users to look at these channels without being
+    /// logged in to Discord.
+    /// Only crossposted messages (via Channel Following) currently include `mention_channels` at all.
+    /// If no mentions in the message meet these requirements, this field will not be sent.
+    /// [`See more in the discord documentation`].
+    ///
+    /// [`See more in the discord documentation`]: https://discord.com/developers/docs/resources/channel#message-object
```
Comment (arqunis) [inline src/model/channel/message.rs]: Content is fine, but the paragraphs can be divided and the text wrapped at around 100 columns better. Also, to allow for changing the text inside the link, I suggest using reference-text. Additionally, to be consistent with the rest of the documentation that recommends users to read other sources of documentation or literature, the link text should use the "Refer to xxx" language.
```suggestion
    ```suggestion
    /// **Note**:
    /// Not all channel mentions in a message will appear in `mention_channels`. Only textual 
    /// channels that are visible to everyone in a lurkable guild will ever be included.
    /// 
    /// A lurkable guild is one that allows users to read public channels in a server without
    /// actually joining the server. It also allows users to look at these channels without being
    /// logged in to Discord.
    ///
    /// Only crossposted messages (via Channel Following) currently include `mention_channels` at 
    /// all. If no mentions in the message meet these requirements, this field will not be sent.
    /// [Refer to Discord's documentation for more information][discord-docs].
    ///
    /// [discord-docs]: https://discord.com/developers/docs/resources/channel#message-object
```
```

**Review by arqunis — CHANGES_REQUESTED** [review]

File: src/model/channel/message.rs (hunk 8 above)
Comment (zacck-zz) [reply src/model/channel/message.rs]: Ah this looks way neater, will commit it

File: src/model/channel/message.rs [hunk 9]
```diff
@@ -72,6 +72,21 @@ pub struct Message {
     /// Array of [`Role`]s' Ids mentioned in the message.
     pub mention_roles: Vec<RoleId>,
     /// Channels specifically mentioned in this message.
+    ///
+    ```suggestion
```
Comment (arqunis) [inline src/model/channel/message.rs]:
```suggestion

```

**Review by arqunis — CHANGES_REQUESTED** [review]

**Review by arqunis — APPROVED** [review]
It is good now. Before I am going to merge this however, you will have to run `cargo fmt --all` at the root directory to pass the format check (assuming you use rustup, be sure you use the Nightly toolchain of Rust and that you have `rustfmt` installed as a component).

**Comment by zacck-zz** [comment]
@acdenisSK  I ran the formatter, hopefully this is good to go?

**Review by arqunis — APPROVED** [review]

**Pull request merged**